### DISCUSSION TITLE
Property 'code' does not exist on type 'ClassAttributes<HTMLElement> & HTMLAttributes<HTMLElement> & ExtraProps'

### DISCUSSION THREAD

**Discussion by erayerdin** [remarkjs/remark#1311] [Q&A] [answered]
My Markdown component is as such:

```tsx
    <Markdown
      remarkPlugins={[remarkGfm]}
      components={{
        code: ({ code, language, showLineNumbers }) => (
          <CopyBlock
            text={code}
            language={language}
            showLineNumbers={showLineNumbers}
            wrapLongLines
          />
        ),
      }}
    >
      {children}
    </Markdown>
```

...where I try to destructure props by `code`, `language` and `showLineNumbers`, but Typescript complains none of them is on the type. 

I'm aware of the removal of `inline` prop. Does the same apply for `code`, `language` and `showLineNumbers`?

I use `react-code-blocks`. How do I render code block with it?

Thanks in advance.

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @erayerdin!
Sorry you ran into a spot of trouble.

I assume you are running `react-markdown` on version 9?
If TypeScript is correctly erroring, none of those properties exist, the actual properties are https://github.com/remarkjs/react-markdown?tab=readme-ov-file#components

For a better sense of what you want to achieve, see the code highlighting example https://github.com/remarkjs/react-markdown?tab=readme-ov-file#use-custom-components-syntax-highlight and map those properties into your code highlighting component of choice.
